include/envoy/secret/dynamic_secret_provider.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/secret/secret_callbacks.h"
+#include "envoy/ssl/tls_certificate_config.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * An interface to fetch dynamic secret.
+ */
+class DynamicSecretProvider {
```

Comment:
I'm curious why we need this and `createDynamicSecretProvider`, but not make SecretManager to manage callbacks.